lib/CSSStyleDeclaration.js

```diff
@@ -360,6 +327,45 @@ class CSSStyleDeclaration {
     }
     this[property] = value;
   }
+
+  /**
+   * @param {string} property
+   */
+  removeProperty(property) {
+    if (this._readonly) {
+      const msg = `Property ${property} can not be modified.`;
+      const name = "NoModificationAllowedError";
+      throw new this._global.DOMException(msg, name);
+    }
+    if (!this._values.has(property)) {
+      return "";
+    }
+    const prevValue = this._values.get(property);
+    this._values.delete(property);
+    this._priorities.delete(property);
+    const index = Array.prototype.indexOf.call(this, property);
+    if (index >= 0) {
+      Array.prototype.splice.call(this, index, 1);
+      if (typeof this._onChange === "function") {
+        this._onChange(this.cssText);
+      }
+    }
+    return prevValue;
+  }
+
+  get parentRule() {
+    return this._parentRule;
+  }
+
+  // Non-standard
+  setOptions(opt = {}) {
```

Comment:
Can we add a comment for what this will be used for? (In general, changing options after creation seems unusual, so documenting it is good.)

Reply:
I think I am still confused. I tried reading the constructor and cannot find any documentation on what options / styleOpts / styleOptions is going to contain. I guess it is passed through to the parser code?

Do we really need that level of configurability from the outside? If I understand correctly, there are two possible types of `CSSStyleDeclaration`: the ones returned from the `style` getter, and the ones returned from `getComputedStyle()`. Could we just have those two modes be options passed from the outside in the constructor, and then the code in this file determines how to set up the style options passed to the parser code?

In other words, why do we need to change this over the lifetime of the `CSSStyleDeclaration` object, instead of setting it once at creation?